### ISSUE TITLE
Crashes/green screen when enable hardware acceleration on Amlogic devices

### ISSUE THREAD

**Issue by rawad1989** [wang-bin/fvp#56]
**Describe the bug**
When trying to enable hardware acceleration with the following regirestion parameter app crashes on MediaCodec_loop loop
  registerWith(options: {
    'video.decoders': ['FFmpeg:codec=h264_mediacodec:hwaccel=mediacodec:hwcontext=mediacodec'],
    'global':{
      'logLevel': 'All',
      'profiler.gpu': 1,
      'ffmpeg.log': 'verbose'
    }
  },);

**Expected behavior**

**Log**
I/flutter (31094): mdk.INFO: 01:34:51.333: [FFmpeg:h264_mediacodec] Output crop parameters top=0 bottom=1079 left=0 right=1919, resulting dimensions width=1920 height=1080
I/flutter (31094): mdk.INFO: 01:34:51.334: [FFmpeg:h264_mediacodec] MediaCodec started successfully: codec = OMX.amlogic.avc.decoder.awesome2, ret = 0
I/flutter (31094): mdk.INFO: 01:34:51.335: AVCodec.Video[h264_mediacodec.] decoder: 1 Single threads
I/flutter (31094): mdk.INFO: 01:34:51.335: unsupported avcodec options:  codec=h264_mediacodec hwaccel=mediacodec hwcontext=mediacodec
I/flutter (31094): mdk.INFO: 01:34:51.336: h264 unknown, bpc:0, bpp:0(), channels:(), map: 0 0 0 0 1920x1080
I/flutter (31094): mdk.INFO: 01:34:51.336: decode stored recovery packets: 0
D(31094): PlayerBase::stop() from IPlayer
D/AudioTrack(31094): stop(57): called with 3712 frames delivered
D/(31094): PlayerBase::stop() from IPlayer
D/AudioTrack(31094): stop(57): called with 128 frames delivered
I/flutter (31094): mdk.INFO: 01:34:51.383: >334 14:00:13.5/00:00:00 cache 1.9s/3417KB 909KB/s
D/(31094): PlayerBase::stop() from IPlayer
D/AudioTrack(31094): stop(57): called with 256 frames delivered
D/SurfaceUtils(31094): set up nativeWindow 0xe417bd38 for 1920x1080, color 0x11, rotation 0, usage 0xc02b00
W/ACodec  (31094): [OMX.amlogic.avc.decoder.awesome2] setting nBufferCountActual to 17 failed: -1010
D/(31094): PlayerBase::stop() from IPlayer
D/AudioTrack(31094): stop(57): called with 3712 frames delivered
D/(31094): PlayerBase::stop() from IPlayer
D/AudioTrack(31094): stop(57): called with 256 frames delivered
D/(31094): PlayerBase::stop() from IPlayer
D/AudioTrack(31094): stop(57): called with 3712 frames delivered
D/(31094): PlayerBase::stop() from IPlayer
D/AudioTrack(31094): stop(57): called with 128 frames delivered
I/flutter (31094): mdk.INFO: 01:34:51.544: >502 14:00:13.6/00:00:00 cache 2.1s/3790KB 9189KB/s
I/flutter (31094): mdk.INFO: 01:34:51.549: [FFmpeg:h264_mediacodec] Output MediaFormat changed to {crop-right=1919, color-format=17, slice-height=1080, mime=video/raw, hdr-static-info=java.nio.HeapByteBuffer[pos=0 lim=25 cap=25], stride=1920, color-standard=1, color-transfer=3, crop-bottom=1079, crop-left=0, width=1920, color-range=2, crop-top=0, height=1080}
I/flutter (31094): mdk.INFO: 01:34:51.555: [FFmpeg:h264_mediacodec] Output crop parameters top=0 bottom=1079 left=0 right=1919, resulting dimensions width=1920 height=1080
F/libc    (31094): Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 31341 (MediaCodec_loop), pid 31094 

Build fingerprint: 'Droidlogic/oppen/oppen:11/RD2A.211001.002/eng.frank.20231019.105840:user/release-keys'
Revision: '0'
ABI: 'arm'
Timestamp: 2023-12-10 01:34:51-0600

pid: 31094, tid: 31341, name: MediaCodec_loop
uid: 10152
signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
    r0  00000000  r1  00007a6d  r2  00000006  r3  acef5df8
    r4  acef5e0c  r5  acef5df0  r6  00007976  r7  0000016b
    r8  acef5df8  r9  acef5e08  r10 acef5e28  r11 acef5e18
    ip  00007a6d  sp  acef5dc8  lr  e573e8ad  pc  e573e8c0
backtrace:

**Comment by wang-bin** [comment]
the backtrace is missing. I need your full log. What about removing `video.decoders` line?

**Comment by rawad1989** [comment]
@wang-bin Removing `video.decoders` it will work but some hls streams will be slow especially the ones with full hd

**Comment by wang-bin** [comment]
the log looks good. what do you mean about slow, low fps? What about using official implementation by removing `registerWith()`? I guess the bottleneck is opengl rendering, if it's true, I have to add an option to let the decoder output frames directly to surface.

**Comment by rawad1989** [comment]
@wang-bin The issue on android device using amlogic cpu s905y4 the picture is slow and when enabling hardware acceleration the image will be green if you can give me your contact info I will give you all details about this issue

**Comment by wang-bin** [comment]
answer my questions in my previous comment, otherwise i can't help. i can add an option to decode in tunnel mode, the performance should be better

**Comment by rawad1989** [comment]
> the log looks good. what do you mean about slow, low fps? What about using official implementation by removing `registerWith()`? I guess the bottleneck is opengl rendering, if it's true, I have to add an option to let the decoder output frames directly to surface.

Without `registerWith ` I'm getting the same issue, I have tried `registerWith ` in order to try another decoders.
Meanwhile can you stop opengl rendering?

**Comment by wang-bin** [comment]
Use the latest code in master branch and add `'tunnel': true`
```dart
registerWith(options: {
    'tunnel': true,
    'global':{
      'logLevel': 'All',
      'profiler.gpu': 1,
      'ffmpeg.log': 'verbose'
    }
  },);
```

**Comment by rawad1989** [comment]
@wang-bin I'm getting this after sitting tunnel: true
I/flutter ( 9534): mdk.FINEST: 06:23:07.103: no video renderer
I/flutter ( 9534): mdk.FINEST: 06:23:07.103: no video renderer
I/flutter ( 9534): mdk.FINEST: 06:23:07.117: no video renderer
I/flutter ( 9534): mdk.FINEST: 06:23:07.173: no video renderer

**Comment by wang-bin** [comment]
> @wang-bin I'm getting this after sitting tunnel: true
> I/flutter ( 9534): mdk.FINEST: 06:23:07.103: no video renderer
> I/flutter ( 9534): mdk.FINEST: 06:23:07.103: no video renderer
> I/flutter ( 9534): mdk.FINEST: 06:23:07.117: no video renderer
> I/flutter ( 9534): mdk.FINEST: 06:23:07.173: no video renderer
> 

ignore this log, it's expected

**Comment by rawad1989** [comment]
@wang-bin nothing rendered on the screen that's why, it seems to be issue on the amlogic s905y4 chipset

**Comment by wang-bin** [comment]
there is a bug in my dart code, I create decoder too early, sometimes tunnel mode is not applied, I don't have a good solution right now. can you try this example https://github.com/wang-bin/mdk-android with the latest library https://sourceforge.net/projects/mdk-sdk/files/nightly/mdk-sdk-android.7z/download ? also change this line https://github.com/wang-bin/mdk-android/blob/master/sdk/src/main/cpp/MDKPlayerJNI.cpp#L13  to `#define  DECODE_TO_SURFACEVIEW 1`

**Comment by rawad1989** [comment]
@wang-bin Do you mean to compile it under the flutter application?

**Comment by wang-bin** [comment]
no, it's a minimal android example without flutter which can decode to SurfaceView directly. fvp plugin can not perform better than this example. i guess there is bottleneck in flutter render engine, so default flutter video player(exoplayer) result is poor. you can set your video url at this line https://github.com/wang-bin/mdk-android/blob/master/app/src/main/java/com/mediadevkit/mdkplayer/MainActivity.java#L64
